src/test/java/github/weichware10/util/FilesTest.java

```diff
@@ -0,0 +1,53 @@
+package github.weichware10.util;
+
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+/**
+ * Testet {@link Files}.
+ */
+public class FilesTest {
+
+    @Test
+    public void shouldSaveImageFromUrl() {
+        String location1 = null;
+        try {
+            location1 = Files.saveImage(
+                    "https://www.planet-wissen.de/natur/voegel/eulen/eulen-uhu-kopf-100~_v-ARDFotogalerie.jpg");
+        } catch (Exception e) {
+            ;
+        }
+        assertTrue(location1.contains("eulen-uhu-kopf-100~_v-ARDFotogalerie.jpg"));
+
+        String location2 = null;
+        try {
+            location2 = Files.saveImage(
+                    "https://purepng.com/public/uploads/large/purepng.com-owl-sittingowlowletbrown-owlowl-from-the-side-4815210279097vhbr.png");
+        } catch (Exception e) {
+            ;
+        }
+        assertTrue(location2.contains(
+                "purepng.com-owl-sittingowlowletbrown-owlowl-from-the-side-4815210279097vhbr.png"));
+
+        boolean thrown1 = false;
+        try {
+            Files.saveImage("https://www.google.de/");
+        } catch (Exception e) {
+            thrown1 = true;
+        }
+        assertTrue(thrown1);
+
+        boolean thrown2 = false;
+        try {
+            Files.saveImage("https://upload.wikimedia.org/wikipedia/commons/2/2c/Rotating_earth_%28large%29.gif");
+        } catch (Exception e) {
+            thrown2 = true;
+        }
+        assertTrue(thrown2);
+
+        assertTrue(Files.tmpdir.contains("weichware"));
+
+        Runtime.getRuntime().addShutdownHook(Files.deleteTempDir());
```

Comment:
```suggestion
        Files.deleteTempDir().start();
```
könnte man das nicht so machen? da ist es vllt weniger boogy

Reply:
ich versuche es mal

Reply:
![image](https://user-images.githubusercontent.com/82680013/149006063-21a64c57-7629-4a60-a3d7-950c4b437195.png)
Hat tatsächlich geholfen, da jetzt der Thread direkt ausgeführt wird :)

Reply:
mahaha!
und jetzt fällt mir noch was auf, ich push es mal